### ISSUE TITLE
text-field: dense does not shrink the height

### ISSUE THREAD

**Issue by seriema** [material-components/material-components-web-react#805]
Adding the `dense` prop to a text field component doesn't change the visual of the component. The Material guidelines mention it clearly (see "Dense text fields" under ["Filled](https://material.io/design/components/text-fields.html#filled-text-field) and ["Outlined"](https://material.io/design/components/text-fields.html#outlined-text-field)) but the mdc-textfield has deprecated it and it's going to be removed:
![image](https://user-images.githubusercontent.com/693684/55974143-53eea580-5c87-11e9-9822-cdbb319f2cd8.png)
https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield#deprecation-notice

Having that property documented here when it doesn't work is confusing and trying to add it when it's being removed upstream doesn't make sense to me.

**Comment by moog16** [comment]
Yes, however the dense class is still available for devs to use. So we still should be supporting it until MDC Web actually removes it. We can however update our docs to reflect that it may/may not be deprecated.

**Comment by seriema** [comment]
It does not work so then it should be fixed?

**Comment by seriema** [comment]
It's also not available in the demos:
https://material-components.github.io/material-components-web-catalog/#/component/text-field

**Comment by moog16** [comment]
I'm not quite sure what you mean that there is no visual difference. If you add the prop `dense` to the textField component, it addes the --dense class, which updates the styles. This can be seen in the stackblitz [here](https://stackblitz.com/edit/react-eeyij6?file=TextField.js).

**Comment by seriema** [comment]
By looking at these two text-fields, one cannot see which one has `dense={true}`:
![image](https://user-images.githubusercontent.com/693684/56655715-df0e6900-6693-11e9-9548-d66c2d1719c2.png)
https://stackblitz.com/edit/text-field-dense?file=TextField.js

The font changes slightly on the placeholder label, but the text field height remains the same. `dense` having the same height isn't what I'd expect. I could write my own CSS, but I haven't needed to do that for other props. So just updating the docs to set the right expectations would help.

**Comment by moog16** [comment]
The MDC web team has been talking about deprecating dense altogether. So I'm fine removing dense. We can keep this issue open, however I'm not sure when we will prioritize it. Please feel free to open a PR until then.
